providers/go-feature-flag-in-process/pkg/provider.go

```diff
@@ -0,0 +1,198 @@
+package gofeatureflaginprocess
+
+import (
+	"context"
+	"fmt"
+	of "github.com/open-feature/go-sdk/openfeature"
+	ff "github.com/thomaspoignant/go-feature-flag"
+	"github.com/thomaspoignant/go-feature-flag/ffcontext"
+)
+
+// Provider is the OpenFeature provider for GO Feature Flag.
+type Provider struct {
+	// goFeatureFlagInstance is the instance of the GO Feature Flag module.
+	goFeatureFlagInstance *ff.GoFeatureFlag
+}
+
+// NewProvider allows you to create a GO Feature Flag provider without any context.
+// We recommend using the function NewProviderWithContext and provide your context when creating the provider.
+func NewProvider(options ProviderOptions) (*Provider, error) {
+	return NewProviderWithContext(context.Background(), options)
+}
+
+// NewProviderWithContext is the easiest way of creating a new GO Feature Flag provider.
+func NewProviderWithContext(ctx context.Context, options ProviderOptions) (*Provider, error) {
+	if options.GOFeatureFlagConfig == nil {
+		return nil, fmt.Errorf("invalid provider options, empty GOFeatureFlagConfig value")
+	}
+
+	goff, err := ff.New(*options.GOFeatureFlagConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &Provider{
+		goFeatureFlagInstance: goff,
```

Comment:
So these are the same internals that power GoFF as a standalone server I guess? That's cool!

Do you plan to support in-process in other languages? I guess that would be a lot harder :sweat_smile:

Reply:
Yes exactly, this is the complete GO Feature Flag code behind the provider.

In-process for other languages is something I have in mind, but I haven't found the right approach for now, the rule library I use is mainly for GO only, and I am not sure how much effort it is to have the same compatibility level for other languages.